control/statesp.py

```diff
@@ -612,14 +612,20 @@ def dcgain(self):
             at the origin
         """
         try:
-            gain = np.asarray(self.D -
-                              self.C.dot(np.linalg.solve(self.A, self.B)))
+            if self.isctime():
+                gain = np.asarray(self.D -
+                                    self.C.dot(np.linalg.solve(self.A, self.B)))
+            else:
+                gain = self.horner(1)
         except LinAlgError:
-            # zero eigenvalue: singular matrix
-            return np.nan
+            # eigenvalue at DC
+            # TODO: should this be of shape (outputs,inputs) ?
+            # TODO: inf rather than nan?
+            # TODO: could one not do a bit better, maybe by finding a least squares solution?
+            #       consider G(s) = diag(1/s, 1/(s+1)); DC gain is [[inf, 0],[0,1]]
+            gain = np.nan
         return np.squeeze(gain)
 
-
```

Comment:
blank line deletion here does not seem to be relevant to the other changes in this commit.